### PR TITLE
Ensuring that hbar can only be specified in `Settings`

### PR DESCRIPTION
**Pull request by SamFerracin**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Ensure that code and tests are properly formatted, by running `make format` or `black -l 100
      <filename>` on any relevant files. You will need to have the Black code format installed:
      `pip install black`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The Mr Mustard source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/) except line length.
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint mrmustard/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
While the value of `hbar` is specified in `Settings`, some methods allow passing (potentially different) values of `hbar`. There is no mechanism to keep track of the different values of `hbar` employed in a simulation, nor to ensure consistency. This leaves room for unintended behaviors.
 
**Description of the Change:**
All methods retrieve the value of `hbar` from the `Settings`, instead of allowing users to specify it in the inputs.

**Benefits:**
The value of `hbar` is guaranteed to be consistent throughout the session, meaning that at any point, it is equal to the value that can be found in the `Settings`.

**Possible Drawbacks:**
Users may still change the `Settings` (and specifically, the value of `hbar`) at any point during the computation. This will be disallowed in a follow-up Pull Request.

**Related GitHub Issues:**

### CODE REVIEW

File: mrmustard/utils/wigner.py [hunk 1]
```diff
@@ -30,13 +29,18 @@ def wigner_discretized(rho, qvec, pvec, hbar=settings.HBAR):
...
     Retunrs:
         tuple(array, array, array): array containing the discretized Wigner function, and the Q and
             P coordinates (in meshgrid form) in which the function is calculated
     """
+    hbar = settings.HBAR
+    return _wigner_discretized(rho, qvec, pvec, hbar)
+
 
+@njit
+def _wigner_discretized(rho, qvec, pvec, hbar):
```
Comment (SamFerracin) [inline mrmustard/utils/wigner.py:42]: @sylviemonet just a comment about this:

The best thing would be to have 
`@njit`
`def wigner_discretized(rho, qvec, pvec)`:
`    hbar = settings.HBAR`
`    ...`
But unfortunately, `@njit` dislikes `settings.HBAR`. 

This is why I decided to go through the `_wigner_discretized` function, which does take `hbar` as input, but is hidden and won't be used anywhere else. If you have better ideas, ket me know :)

File: tests/test_physics/test_fock/test_fock.py [hunk 2]
```diff
@@ -37,6 +37,10 @@
 from mrmustard.math.lattice.strategies import displacement, grad_displacement
 from mrmustard.physics import fock
 
+from mrmustard import settings
+
+settings.HBAR = 2
```
Comment (ziofil) [inline tests/test_physics/test_fock/test_fock.py:42]: why do we need this here?

File: tests/test_physics/test_gaussian/test_symplectics.py [hunk 3]
```diff
@@ -34,6 +40,10 @@
 from mrmustard.lab.states import TMSV, Thermal, Vacuum
 from mrmustard.physics.gaussian import controlled_X, controlled_Z
 
+from mrmustard import settings
+
+settings.HBAR = 2
```
Comment (ziofil) [inline tests/test_physics/test_gaussian/test_symplectics.py:45]: I guess it's for the same reason, but why do we need to set hbar here?

**Review by ziofil — APPROVED** [review]
Great job! I only left a couple comments

File: tests/test_physics/test_gaussian/test_symplectics.py (hunk 3 above)
Comment (SamFerracin) [reply tests/test_physics/test_gaussian/test_symplectics.py:45]: Thanks for spotting this, I had forgotten it there

File: mrmustard/lab/states.py [hunk 4]
```diff
@@ -421,7 +426,10 @@ def __init__(
         eigenvalues: Vector = None,
         symplectic_trainable: bool = False,
         eigenvalues_trainable: bool = False,
-        eigenvalues_bounds: Tuple[Optional[float], Optional[float]] = (settings.HBAR / 2, None),
+        eigenvalues_bounds: Tuple[Optional[float], Optional[float]] = (
+            settings.HBAR / 2,
```
[the remark this answers is not in the document]
Comment (SamFerracin) [reply mrmustard/lab/states.py:430]: I was a bit confused here. Should the lower bound always be hbar/2, or can I replace it with `None`?

**Pull request merged**